File: onlineshop4dvds_api/Entities/Order.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using OnlineShop4DVDS.Utils;

namespace OnlineShop4DVDS.Entities;

public class Order
{
    [Key]
    public int Id {get;set;}

    public DateTime CreatedAt {get;set;} = DateTime.UtcNow;

    public required string OrderId {get;set;}

    public required OrderStatus Status {get;set;}

    public required decimal Subtotal {get;set;}
    public required decimal ShippingFee {get;set;}
    public decimal? Discount {get;set;}

    [ForeignKey(nameof(User))]
    public required int UserId {get;set;}
    public User? User {get;set;}

    public ICollection<Product>? Products {get;set;}
    public ICollection<OrderProduct>? OrderProducts {get;set;}

    public required PaymentMethod PaymentMethod {get;set;}
}

File: onlineshop4dvds_app/src/models/order.ts
import { OrderStatus } from "@/utils/order";
import { PaymentMethod } from "@/utils/payment";

export interface OrderCreate {
    userSub: string;
    orderId: string;
    shippingFee: number;
    discount: number;
    paymentMethod: PaymentMethod
    items: OrderItemCreate[];
}

export interface OrderItemCreate {
    productId: number;
    quantity: number;
}

export interface Order {
   id: number;
   createdAt: string;
   orderId: string;
   status: OrderStatus;
   subtotal: number;
   shippingFee: number;
   discount: number;
   paymentMethod: string;
   items: OrderItem[];
}

export interface OrderItem {
    title: string;
    price: number;
    quantity: number;
    thumbnailUrl: string;
    type: string;
    productId: number;
}
